tsdb/chunks/chunks.go

```diff
@@ -262,7 +264,10 @@ func MergeOverlappingChunks(chks []Meta) ([]Meta, error) {
 		}
 		nc.Chunk = chk
 	}
-
+	newChks, err := BreakLargeChunks(newChks, MaxSamplesInChunk)
```

Comment:
Nice! All looks good but I am pretty certain it would be more efficient do it in `MergeChunks` instead. We can reuse exactly the same loop iteration (:

Reply:
Hello bwplotka. Thanks for taking the time to review my code.
Can you please provide some more added clarification about your comment here?
I think what you have pointed out is that we are iterating through the []Meta once to just do the `MergeChunks` and then once again inside the function I wrote. Is your proposal to change this nature of the code? 

Including my function within `MergeChunks` would require changes to the function signature itself. It is a public function and I'm not too sure if thats what we want to achieve here.

Reply:
> I think what you have pointed out is that we are iterating through the []Meta once to just do the MergeChunks and then once again inside the function I wrote. Is your proposal to change this nature of the code?

Yes.

> Including my function within MergeChunks would require changes to the function signature itself. It is a public function and I'm not too sure if thats what we want to achieve here.

This is fine. As Prometheus, we don't guarantee stability of internal package API. All of our users user versioned package of Prometheus (e.g project I maintain: [Thanos](http://thanos.io/)). Someday we might want proper semantic versioning for our packages, but not for now.

So it's totally fine to change signature. If you would silently change logic / behaviour, that would be worse. So go for it :+1:

Reply:
Okay Sure. Your comments have been incorporated in my latest PR. Please do have a look when possible.